### PR TITLE
feat(inx): remove `LedgerUpdateStream` and manual chunks iter

### LINKED ISSUE #689: Remove the `ChunksExt` iterator and use `itertools::Chunks`
With the actor system removed, we probably don't need the `ChunksExt` iterator anymore and can use `itertools::Chunks` instead.

### LINKED ISSUE #702: Remove `LedgerUpdateStream`

### PR DESCRIPTION
**Pull request by Alexandcoats**
## Linked Issues

* Closes #702
* Closes #689 

## Notes to Reviewer

As a reviewer, please pay particular attention to the following areas when reviewing this PR and tick the above boxes after you have completed the steps.

#### INX Changes
* [x] Run chronicle using an INX connection.
* [x] Measure performance differences between previous and new implementation.

### CODE REVIEW

**Comment by grtlr** [comment]
Any reason why you choose not to do the handling similar to: https://github.com/grtlr/nort/blob/main/src/inx.rs#L22-L57? I think doing things this way would allow us to write a generic, iterator-based, chunk inserter that would be easier to test and reuse.

**Comment by grtlr** [comment]
Here we should assume it though because it's guaranteed to be like that by INX: https://github.com/iotaledger/inx/blob/develop/proto/inx.proto#L37-L41

**Comment by Alexandcoats** [comment]
> Here we should assume it though because it's guaranteed to be like that by INX: https://github.com/iotaledger/inx/blob/develop/proto/inx.proto#L37-L41

Sure, but it's not guaranteed by the API. Since it's not an issue to handle it this way, I don't think we should change it.

> I'd much rather use the regular `chunks` method instead of rolling our own.

As I have stated in that issue, it is not possible.

> Also, regarding the iterator implementation, I was refering to this: https://github.com/iotaledger/inx-chronicle/issues/704

We could make something generic, but why? This is used in one place. Also, my impl parallelizes the places that can be.

File: src/bin/inx-chronicle/stardust_inx/mod.rs [hunk 1]
```diff
@@ -75,12 +72,12 @@ impl InxWorker {
...
         let (start_index, mut inx) = self.init().await?;
 
-        let mut stream = LedgerUpdateStream::new(inx.listen_to_ledger_updates((start_index.0..).into()).await?);
+        let mut stream = inx.listen_to_ledger_updates((start_index.0..).into()).await?;
 
         debug!("Started listening to ledger updates via INX.");
 
         while let Some(ledger_update) = stream.try_next().await? {
-            handle_ledger_update(ledger_update, &mut inx, &self.db).await?;
+            handle_ledger_update(&mut inx, &self.db, ledger_update, &mut stream).await?;
```
Comment (Alex6323) [inline src/bin/inx-chronicle/stardust_inx/mod.rs:80]: at this point we can just move the `while let` into `handle_ledger_update` IMO

File: src/bin/inx-chronicle/stardust_inx/mod.rs [hunk 2]
```diff
@@ -213,46 +211,109 @@ impl InxWorker {
...
+        .try_chunks(INSERT_BATCH_SIZE)
+        // We only care if we had an error, so discard the other data
+        .map_err(|e| e.1)
+        // Convert batches to tasks
+        .try_fold(&mut tasks, |tasks, batch| async {
+            let db = db.clone();
+            tasks.spawn(async move { insert_unspent_outputs(&db, &batch).await });
+            Result::<_, InxError>::Ok(tasks)
+        })
+        .await?;
```
Comment (Alex6323) [inline src/bin/inx-chronicle/stardust_inx/mod.rs:279]: make this DRY with a macro?

File: src/bin/inx-chronicle/stardust_inx/mod.rs (hunk 1 above)
Comment (Alexandcoats) [reply src/bin/inx-chronicle/stardust_inx/mod.rs:80]: Well the reason I did it this way is to keep the span around the function.

File: src/bin/inx-chronicle/stardust_inx/mod.rs (hunk 2 above)
Comment (Alexandcoats) [reply src/bin/inx-chronicle/stardust_inx/mod.rs:279]: Oh hell no

Comment (Alex6323) [reply src/bin/inx-chronicle/stardust_inx/mod.rs:279]: :joy:

File: src/bin/inx-chronicle/stardust_inx/mod.rs (hunk 1 above)
Comment (Alex6323) [reply src/bin/inx-chronicle/stardust_inx/mod.rs:80]: oh I see ... damn it!

**Review by Alex6323 — DISMISSED** [review]

**Pull request merged**